aioxmpp/presence/service.py

```diff
@@ -236,6 +236,183 @@ def handle_presence(self, st):
+        """
+        return self._muted
+
+    @property
+    def presence_filter(self):
+        """
+        Optional callback which is invoked on the presence stanza before it is
+        sent.
+
+        This is called whenever a presence stanza is sent for this relationship
+        by the :class:`PresenceServer`. This is not invoked for presence stanzas
+        sent to the peer by other means (e.g. :meth:`aioxmpp.Client.send`).
+
+        If the :attr:`presence_filter` is not :data:`None`, it is called with
+        the presence stanza as its only argument. It must either return the
+        presence stanza, or :data:`None`. If it returns :data:`None`, the
+        presence stanza is not sent.
+
+        The callback operates on a copy of the presence stanza to prevent
+        modifications from leaking into other presence relationships; making a
+        copy inside the callback is not required or recommended.
+        """
+        return self._presence_filter
+
+    @presence_filter.setter
+    def presence_filter(self, new_callback):
+        self._presence_filter = new_callback
+
+    def set_muted(self, muted: bool, *, send_update_now: bool = True):
+        """
+        Change the :attr:`muted` state of the relationship.
+
+        (This is not a setter to the property due to the additional options
+        which are available when changing the :attr:`muted` state.)
+
+        :param muted: The new muted state.
+        :type muted: :class:`bool`
+        :param send_update_now: Whether to send a presence update to the peer
+            immediately.
+        :type send_update_now: :class:`bool`
+        :raises RuntimeError: if the presence relationship has been destroyed
+            with :meth:`unsubscribe`
+
+        If `muted` is equal to :attr:`muted`, this method does nothing.
+
+        If `muted` is :data:`True`, the presence relationship will be muted.
+        `send_update_now` is ignored.
+
+        If `muted` is :data:`False`, the presence relationship will be unmuted.
+        If `send_update_now` is :data:`True`, the current presence is sent to
+        the peer immediately.
+        """
+        self._require_subscribed()
+
+        muted = bool(muted)
+        if muted == self._muted:
+            return
+
+        self._muted = muted
+        self._service._emit_presence_directed(self)
+
+    def unsubscribe(self):
+        """
+        Destroy the directed presence relationship.
+
+        The presence relationship becomes useless afterwards. Any additional
+        calls to the methods will result in :class:`RuntimeError`. No additional
+        updates wil be sent to the peer automatically (except, of course, if
+        a new relationship is created).
+
+        If the presence relationship is still active when the method is called,
+        :attr:`~aioxmpp.PresenceType.UNAVAILABLE` presence is sent to the peer
+        immediately. Otherwise, no stanza is sent. The stanza is passed through
```

Comment:
Do we want this to happen in all cases? (e.g. to peers that are subscribed to our presence).